runtime error on `gopass --generate-bash-completion`

### Summary
gpg went missing from my system due to some weird Homebrew behavior. Then, bash completion stopped worked on `gopass<TAB>`. This triggered calling `gopass --generate-bash-completion`, which exits with a runtime error.

### Steps To Reproduce
```
$ which gpg
gpg not found
$ gopass --generate-bash-completion
agent
audit
binary
bin
clone
completion
config
copy
cp
create
new
delete
remove
rm
edit
set
find
search
fsck
generate
git-credential
otp
totp
hotp
git
grep
history
hist
init
insert
list
ls
move
mv
mounts
recipients
show
sync
templates
update
version
xc
help
h
Faild to initialize stores: failed to initialize the root store at 'gpgcli-gitcli-fs+file:///Users/tagir/git/password-store/root': failed to init crypto backend: no gpg binary found: no gpg binary found: failed to init crypto backend: no gpg binary found: no gpg binary found
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x10ec994fa]

goroutine 1 [running]:
github.com/gopasspw/gopass/pkg/store/sub.(*Store).List(0x0, 0x10f0fe620, 0xc4202ba570, 0x0, 0x0, 0x0, 0xc42016da80, 0x10e82ada7, 0xc420310000, 0xc4202d4200)
	/private/tmp/gopass-20180608-28807-qxx3yy/gopass-1.8.1/src/github.com/gopasspw/gopass/pkg/store/sub/list.go:17 +0x3a
github.com/gopasspw/gopass/pkg/store/root.(*Store).Tree(0xc420082380, 0x10f0fe620, 0xc4202ba570, 0x1e, 0xc420069130, 0x1, 0x1)
	/private/tmp/gopass-20180608-28807-qxx3yy/gopass-1.8.1/src/github.com/gopasspw/gopass/pkg/store/root/list.go:56 +0x1ad
github.com/gopasspw/gopass/pkg/store/root.(*Store).List(0xc420082380, 0x10f0fe620, 0xc4202ba570, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/gopass-20180608-28807-qxx3yy/gopass-1.8.1/src/github.com/gopasspw/gopass/pkg/store/root/list.go:18 +0x43
github.com/gopasspw/gopass/pkg/action.(*Action).Complete(0xc4202e0fc0, 0x10f0fe620, 0xc4202ba570, 0xc42031e000)
	/private/tmp/gopass-20180608-28807-qxx3yy/gopass-1.8.1/src/github.com/gopasspw/gopass/pkg/action/completion.go:30 +0x96
main.setupApp.func1(0xc42031e000)
	src/github.com/gopasspw/gopass/app.go:58 +0x72
github.com/gopasspw/gopass/vendor/github.com/urfave/cli.ShowCompletions(0xc42031e000)
	/private/tmp/gopass-20180608-28807-qxx3yy/gopass-1.8.1/src/github.com/gopasspw/gopass/vendor/github.com/urfave/cli/help.go:224 +0x49
github.com/gopasspw/gopass/vendor/github.com/urfave/cli.checkCompletions(0xc42031e000, 0xc42031e000)
	/private/tmp/gopass-20180608-28807-qxx3yy/gopass-1.8.1/src/github.com/gopasspw/gopass/vendor/github.com/urfave/cli/help.go:334 +0x7c
github.com/gopasspw/gopass/vendor/github.com/urfave/cli.(*App).Run(0xc4202f8000, 0xc4200c2000, 0x1, 0x2, 0x0, 0x0)
	/private/tmp/gopass-20180608-28807-qxx3yy/gopass-1.8.1/src/github.com/gopasspw/gopass/vendor/github.com/urfave/cli/app.go:206 +0x351
main.main()
	src/github.com/gopasspw/gopass/main.go:64 +0x2d7
```

### Expected behavior
The completion to give back the possible arguments.

### Environment
- OS: macOS High Sierra 10.13.5
- OS version: `Darwin mbp-tba 17.6.0 Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64 x86_64`
- gopass Version: 1.8.1
- Installation method: brew

--- a/pkg/action/clone.go
+++ b/pkg/action/clone.go
@@ -46,7 +46,11 @@ func (s *Action) clone(ctx context.Context, repo, mount, path string) error {
 	if path == "" {
 		path = config.PwStoreDir(mount)
 	}
-	if mount == "" && s.Store.Initialized(ctx) {
+	inited, err := s.Store.Initialized(ctx)
+	if err != nil {
+		return ExitError(ctx, ExitUnknown, err, "Failed to initialized stores: %s", err)
+	}
+	if mount == "" && inited {
 		return ExitError(ctx, ExitAlreadyInitialized, nil, "Can not clone %s to the root store, as this store is already initialized. Please try cloning to a submount: `%s clone %s sub`", repo, s.Name, repo)
 	}
 
@@ -109,7 +113,11 @@ func (s *Action) cloneAddMount(ctx context.Context, mount, path string) error {
 		return nil
 	}
 
-	if !s.Store.Initialized(ctx) {
+	inited, err := s.Store.Initialized(ctx)
+	if err != nil {
+		return ExitError(ctx, ExitUnknown, err, "Failed to initialize store: %s", err)
+	}
+	if !inited {
 		return ExitError(ctx, ExitNotInitialized, nil, "Root-Store is not initialized. Clone or init root store first")
 	}
 	if err := s.Store.AddMount(ctx, mount, path); err != nil {

--- a/pkg/action/completion.go
+++ b/pkg/action/completion.go
@@ -8,6 +8,7 @@ import (
 
 	fishcomp "github.com/gopasspw/gopass/pkg/completion/fish"
 	zshcomp "github.com/gopasspw/gopass/pkg/completion/zsh"
+	"github.com/gopasspw/gopass/pkg/out"
 
 	"github.com/urfave/cli"
 )
@@ -26,7 +27,11 @@ func bashEscape(s string) string {
 
 // Complete prints a list of all password names to os.Stdout
 func (s *Action) Complete(ctx context.Context, c *cli.Context) {
-	s.Store.Initialized(ctx) // important to make sure the structs are not nil
+	_, err := s.Store.Initialized(ctx) // important to make sure the structs are not nil
+	if err != nil {
+		out.Red(ctx, "Store not initialized: %s", err)
+		return
+	}
 	list, err := s.Store.List(ctx, 0)
 	if err != nil {
 		return

--- a/pkg/action/generate.go
+++ b/pkg/action/generate.go
@@ -262,7 +262,11 @@ func (s *Action) CompleteGenerate(ctx context.Context, c *cli.Context) {
 	}
 	needle := args[0]
 
-	s.Store.Initialized(ctx) // important to make sure the structs are not nil
+	_, err := s.Store.Initialized(ctx) // important to make sure the structs are not nil
+	if err != nil {
+		out.Red(ctx, "Store not initialized: %s", err)
+		return
+	}
 	list, err := s.Store.List(ctx, 0)
 	if err != nil {
 		return

--- a/pkg/action/init.go
+++ b/pkg/action/init.go
@@ -23,19 +23,25 @@ import (
 // Initialized returns an error if the store is not properly
 // prepared.
 func (s *Action) Initialized(ctx context.Context, c *cli.Context) error {
-	if !s.Store.Initialized(ctx) {
-		out.Debug(ctx, "Store needs to be initialized")
-		if !ctxutil.IsInteractive(ctx) {
-			return ExitError(ctx, ExitNotInitialized, nil, "password-store is not initialized. Try '%s init'", s.Name)
-		}
-		if ok, err := termio.AskForBool(ctx, "It seems you are new to gopass. Do you want to run the onboarding wizard?", true); err == nil && ok {
-			if err := s.InitOnboarding(ctx, c); err != nil {
-				return ExitError(ctx, ExitUnknown, err, "failed to run onboarding wizard: %s", err)
-			}
-			return nil
-		}
+	inited, err := s.Store.Initialized(ctx)
+	if err != nil {
+		return ExitError(ctx, ExitUnknown, err, "Failed to initialize store: %s", err)
 	}
-	out.Debug(ctx, "Store is already initialized")
+	if inited {
+		out.Debug(ctx, "Store is already initialized")
+		return nil
+	}
+
+	out.Debug(ctx, "Store needs to be initialized")
+	if !ctxutil.IsInteractive(ctx) {
+		return ExitError(ctx, ExitNotInitialized, nil, "password-store is not initialized. Try '%s init'", s.Name)
+	}
+	if ok, err := termio.AskForBool(ctx, "It seems you are new to gopass. Do you want to run the onboarding wizard?", true); err == nil && ok {
+		if err := s.InitOnboarding(ctx, c); err != nil {
+			return ExitError(ctx, ExitUnknown, err, "failed to run onboarding wizard: %s", err)
+		}
+		return nil
+	}
 	return nil
 }
 
@@ -45,7 +51,11 @@ func (s *Action) Init(ctx context.Context, c *cli.Context) error {
 	alias := c.String("store")
 
 	ctx = initParseContext(ctx, c)
-	if s.Store.Initialized(ctx) {
+	inited, err := s.Store.Initialized(ctx)
+	if err != nil {
+		return ExitError(ctx, ExitUnknown, err, "Failed to initialized store: %s", err)
+	}
+	if inited {
 		out.Red(ctx, "WARNING: Store is already initialized")
 	}
 

--- a/pkg/jsonapi/api_test.go
+++ b/pkg/jsonapi/api_test.go
@@ -18,6 +18,7 @@ import (
 	"github.com/gopasspw/gopass/pkg/store/secret"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 type storedSecret struct {
@@ -226,7 +227,9 @@ func runRespondRawMessages(t *testing.T, requests []verifiedRequest, secrets []s
 		},
 	)
 	assert.NoError(t, err)
-	assert.Equal(t, false, store.Initialized(ctx))
+	inited, err := store.Initialized(ctx)
+	require.NoError(t, err)
+	assert.Equal(t, false, inited)
 	assert.NoError(t, populateStore(tempdir, secrets))
 
 	for _, request := range requests {

--- a/pkg/store/root/init.go
+++ b/pkg/store/root/init.go
@@ -13,15 +13,14 @@ import (
 )
 
 // Initialized checks on disk if .gpg-id was generated and thus returns true.
-func (r *Store) Initialized(ctx context.Context) bool {
+func (r *Store) Initialized(ctx context.Context) (bool, error) {
 	if r.store == nil {
 		out.Debug(ctx, "initializing store and possible sub-stores")
 		if err := r.initialize(ctx); err != nil {
-			out.Red(ctx, "Faild to initialize stores: %s", err)
-			return false
+			return false, errors.Wrapf(err, "failed to initialized stores: %s", err)
 		}
 	}
-	return r.store.Initialized(ctx)
+	return r.store.Initialized(ctx), nil
 }
 
 // Init tries to initialize a new password store location matching the object

--- a/pkg/store/root/init_test.go
+++ b/pkg/store/root/init_test.go
@@ -11,6 +11,7 @@ import (
 	"github.com/gopasspw/gopass/tests/gptest"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestInit(t *testing.T) {
@@ -27,8 +28,13 @@ func TestInit(t *testing.T) {
 	rs, err := New(ctx, cfg)
 	assert.NoError(t, err)
 
-	assert.Equal(t, false, rs.Initialized(ctx))
+	inited, err := rs.Initialized(ctx)
+	require.NoError(t, err)
+	assert.Equal(t, false, inited)
 	assert.NoError(t, rs.Init(ctx, "", u.StoreDir("rs"), "0xDEADBEEF"))
-	assert.Equal(t, true, rs.Initialized(ctx))
+
+	inited, err = rs.Initialized(ctx)
+	require.NoError(t, err)
+	assert.Equal(t, true, inited)
 	assert.NoError(t, rs.Init(ctx, "rs2", u.StoreDir("rs2"), "0xDEADBEEF"))
 }

--- a/pkg/store/root/store_test.go
+++ b/pkg/store/root/store_test.go
@@ -131,6 +131,8 @@ func createRootStore(ctx context.Context, u *gptest.Unit) (*Store, error) {
 	if err != nil {
 		return nil, err
 	}
-	s.Initialized(ctx)
+	if _, err := s.Initialized(ctx); err != nil {
+		return nil, err
+	}
 	return s, nil
 }

--- a/pkg/store/sub/list.go
+++ b/pkg/store/sub/list.go
@@ -14,6 +14,10 @@ var (
 
 // List will list all entries in this store
 func (s *Store) List(ctx context.Context, prefix string) ([]string, error) {
+	if s.storage == nil || s.crypto == nil {
+		return nil, nil
+	}
+
 	lst, err := s.storage.List(ctx, prefix)
 	if err != nil {
 		return nil, err

--- a/tests/completion_test.go
+++ b/tests/completion_test.go
@@ -1,6 +1,7 @@
 package tests
 
 import (
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -38,4 +39,19 @@ complete -F _gopass_bash_autocomplete gopass`
 	out, err = ts.run("completion fish")
 	assert.NoError(t, err)
 	assert.Contains(t, out, "complete")
+}
+
+func TestCompletionNoPath(t *testing.T) {
+	ts := newTester(t)
+	defer ts.teardown()
+
+	ov := os.Getenv("PATH")
+	assert.NoError(t, os.Setenv("PATH", "/tmp/foobar"))
+	defer func() {
+		_ = os.Setenv("PATH", ov)
+	}()
+
+	out, err := ts.run("--generate-bash-completion")
+	assert.NoError(t, err)
+	assert.Contains(t, out, "Store not initialized")
 }

